Sources/ManageSettings.php

```diff
@@ -2275,4 +2281,293 @@ function ModifyAlertsSettings()
 	$context['sub_template'] = 'alert_configuration';
 }
 
+/**
+ * Config array for changing privacy settings
+ * Accessed  from ?action=admin;area=featuresettings;sa=privacy;
+ *
+ * @param bool $return_config Whether or not to return the config_vars array
+ * @return void|array Returns nothing or returns the $config_vars array if $return_config is true
+ */
+function ModifyPrivacySettings($return_config = false)
+{
+	global $txt, $scripturl, $context;
+
+	$config_vars = array(
+		array('check', 'enable_privacy_userexport'),
+		array('permissions', 'privacy_userexport_own'),
+		array('permissions', 'privacy_userexport_any'),
+	);
+
+	call_integration_hook('integrate_privacy_settings', array(&$config_vars));
+
+	if ($return_config)
+		return $config_vars;
+
+	// Saving?
+	if (isset($_GET['save']))
+	{
+		checkSession();
+
+		call_integration_hook('integrate_save_privacy_settings');
+
+		saveDBSettings($config_vars);
+		$_SESSION['adm-save'] = true;
+		redirectexit('action=admin;area=featuresettings;sa=privacy');
+	}
+
+	$context['post_url'] = $scripturl . '?action=admin;area=featuresettings;save;sa=privacy';
+	$context['settings_title'] = $txt['privacy'];
+
+	prepareDBSettingContext($config_vars);
+}
+
+/**
+ * Config array for changing policy settings
+ * Accessed  from ?action=admin;area=featuresettings;sa=policy;
+ *
+ * @param bool $return_config Whether or not to return the config_vars array
+ * @return void|array Returns nothing or returns the $config_vars array if $return_config is true
+ */
+function ModifyPolicySettings($return_config = false)
```

Comment:
This function needs some logic to ensure that `enable_policy_function` cannot be set to true unless `policy_version` and any other necessary values are defined. 

Right now, if the admin makes the mistake of enabling the policy function without setting up everything else first, the admin will be redirected on every page load to the page where he is prompted to accept the policy. But the admin will be unable to do so because there is no policy to accept, and so he will be stuck in an endless loop and will be unable to fix his forum except by editing his database directly.

Reply:
Since i check everywhere if enabled_policy_function is true:
https://github.com/SimpleMachines/SMF2.1/blob/66f502c31e0e630d42c3b3a60a56b9345736743b/index.php#L216
I see no need here todo any checks in the settings.